korma/src/commonMain/kotlin/com/soywiz/korma/geom/IRectangle.kt

```diff
@@ -0,0 +1,92 @@
+package com.soywiz.korma.geom
+
+interface IRectangle {
+    val x: Double
+    val y: Double
+    val width: Double
+    val height: Double
+
+    companion object {
+        inline operator fun invoke(
+            x: Double,
+            y: Double,
+            width: Double,
+            height: Double
+        ): IRectangle = Rectangle(x, y, width, height)
+
+        inline operator fun invoke(x: Float, y: Float, width: Float, height: Float): IRectangle =
+            Rectangle(x, y, width, height)
+
+        inline operator fun invoke(x: Int, y: Int, width: Int, height: Int): IRectangle =
+            Rectangle(x, y, width, height)
+
+        // Creates a rectangle from 2 points where the (x,y) is the top left point
+        // with the same width and height as the point.
+        //
+        // Rect XY   Point 2
+        // │        │
+        // ▼        ▼
+        // ┌────────┐
+        // │        │
+        // │        │
+        // └────────┘
+        // ▲
+        // │
```

Comment:
This is a bit misleading, since it says the top left point is 2, but in the diagram it is the bottom left one, and in the implementation it doesnt seems to matter since in the end it uses min and max functions

Reply:
The order or position of the points does not matter, clarified further.